src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformValidator.kt

```diff
@@ -0,0 +1,74 @@
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.transform
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.elasticapi.suspendUntil
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.exceptions.TransformValidationException
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.model.Transform
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.model.TransformValidationResult
+import com.amazon.opendistroforelasticsearch.indexmanagement.util._DOC
+import java.lang.IllegalStateException
+import org.elasticsearch.ExceptionsHelper
+import org.elasticsearch.action.admin.indices.mapping.get.GetMappingsRequest
+import org.elasticsearch.action.admin.indices.mapping.get.GetMappingsResponse
+import org.elasticsearch.action.support.IndicesOptions
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.transport.RemoteTransportException
+
+class TransformValidator(
+    private val indexNameExpressionResolver: IndexNameExpressionResolver,
+    private val clusterService: ClusterService,
+    private val esClient: Client
+) {
+
+    // TODO: When FGAC is supported in transform should check the user has the correct permissions
+    suspend fun validate(transform: Transform): TransformValidationResult {
+        val errorMessage = "Failed to validate the transform job"
+        try {
+            val concreteIndices =
+                indexNameExpressionResolver.concreteIndexNames(clusterService.state(), IndicesOptions.lenientExpand(), transform.sourceIndex)
+            if (concreteIndices.isEmpty()) return TransformValidationResult(false, listOf("No specified source index exist in the cluster"))
+
+            val issues = mutableListOf<String>()
+            concreteIndices.forEach { index -> issues.addAll(validateIndex(index, transform)) }
+
+            return TransformValidationResult(issues.isEmpty(), issues)
+        } catch (e: RemoteTransportException) {
+            val unwrappedException = ExceptionsHelper.unwrapCause(e) as Exception
+            throw TransformValidationException(errorMessage, unwrappedException)
+        } catch (e: Exception) {
+            throw TransformValidationException(errorMessage, e)
+        }
+    }
+
+    private suspend fun validateIndex(index: String, transform: Transform): List<String> {
```

Comment:
Add some kdocs to explain what this is doing

Reply:
will add